src/data/multiset/nodup.lean

```diff
@@ -30,19 +30,16 @@ quot.lift_on s nodup (λ s t p, propext p.nodup_iff)
 @[simp] theorem nodup_cons {a : α} {s : multiset α} : nodup (a ::ₘ s) ↔ a ∉ s ∧ nodup s :=
 quot.induction_on s $ λ l, nodup_cons
 
-theorem nodup_cons_of_nodup {a : α} {s : multiset α} (m : a ∉ s) (n : nodup s) : nodup (a ::ₘ s) :=
-nodup_cons.2 ⟨m, n⟩
+lemma nodup.cons (m : a ∉ s) (n : nodup s) : nodup (a ::ₘ s) := nodup_cons.2 ⟨m, n⟩
 
 theorem nodup_singleton : ∀ a : α, nodup ({a} : multiset α) := nodup_singleton
 
-theorem nodup_of_nodup_cons {a : α} {s : multiset α} (h : nodup (a ::ₘ s)) : nodup s :=
-(nodup_cons.1 h).2
+lemma nodup.of_cons (h : nodup (a ::ₘ s)) : nodup s := (nodup_cons.1 h).2
 
-theorem not_mem_of_nodup_cons {a : α} {s : multiset α} (h : nodup (a ::ₘ s)) : a ∉ s :=
-(nodup_cons.1 h).1
+theorem nodup.not_mem (h : nodup (a ::ₘ s)) : a ∉ s := (nodup_cons.1 h).1
```

Comment:
This is too cryptic for me, how about:
```suggestion
theorem nodup.head_not_mem_tail (h : nodup (a ::ₘ s)) : a ∉ s := (nodup_cons.1 h).1
```

Reply:
I've thought about it a lot and I figured out that `nodup.not_mem` could not mean anything else. I'm happy to change the name but `head` and `tail` already mean something so I'm a bit wary of reusing the names like this.

Reply:
Can't think of a better alternative that reflects the syntax, so let's just go with `not_mem`.